Functions/ImageProcessing/xASL_im_MergePWI4D.m

```diff
@@ -121,15 +121,20 @@
 		for iField2Merge = 1:length(listFields2Merge)
 			% The field exists and is not empty
  			if isfield(jsonPWI4Dcurrent, listFields2Merge{iField2Merge}) && ~isempty(jsonPWI4Dcurrent.(listFields2Merge{iField2Merge}))
-				if iSession == 1 && ~isfield(jsonPWI4DConcatenated, listFields2Merge{iField2Merge})
-					% If non-existing, just add the field
+				if iSession == 1 
+					% For the first session, just add the field
 					jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}) = jsonPWI4Dcurrent.(listFields2Merge{iField2Merge});
-				else
-					% Otherwise check for consistency and report an error if non-consistent
-					if ~isequal(jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}), jsonPWI4Dcurrent.(listFields2Merge{iField2Merge}))
-						error(['Cannot merge JSON field ' listFields2Merge{iField2Merge} ' as it differs between sessions']);
-					end
+				elseif ~isfield(jsonPWI4DConcatenated, listFields2Merge{iField2Merge})
+					% Session > 1, but the field was not added for session 1
+					% We report a conflict
+					error(['Cannot merge JSON field ' listFields2Merge{iField2Merge} ' as it was missing for the first session']);
+				elseif ~isequal(jsonPWI4DConcatenated.(listFields2Merge{iField2Merge}), jsonPWI4Dcurrent.(listFields2Merge{iField2Merge}))
+					% Otherwise (field added for first session and we are now adding more) check for consistency and report an error if non-consistent
+					error(['Cannot merge JSON field ' listFields2Merge{iField2Merge} ' as it differs between sessions']);
```

Comment:
These warnings should be:
'cannot merge ' path ' with ' path ' because field ' jsonField ' differs between sessions.
(or replace path by session)

because in the case of ~bConcatenate, parameters need to be equal and you don't merge parameters, you merge the sessions.

Reply:
Done.